codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -99,9 +105,9 @@ public Decoder(int maxHeadersLength, int maxHeaderTableSize, int initialHuffmanD
      * <p>
      * This method assumes the entire header block is contained in {@code in}.
      */
-    public void decode(ByteBuf in, Http2Headers headers) throws Http2Exception {
+    public void decode(int streamId, ByteBuf in, Http2Headers headers) throws Http2Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "decode" is 49 which is greater than 25 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)